deployment/common/test/changeset/capability_registry.go

```diff
@@ -0,0 +1,73 @@
+package changeset
+
+import (
+	"fmt"
+	"testing"
+
+	chainsel "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/common/view/v1_0"
+	"github.com/smartcontractkit/chainlink/deployment/keystone"
+	"github.com/smartcontractkit/chainlink/deployment/keystone/changeset"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
+)
+
+type HydrateConfig struct {
+	ChainID uint64
+}
+
+// HydrateCapabilityRegistry deploys a new capabilities registry contract and hydrates it with the provided data.
+func HydrateCapabilityRegistry(t *testing.T, v v1_0.CapabilityRegistryView, env deployment.Environment, cfg HydrateConfig) (*capabilities_registry.CapabilitiesRegistry, error) {
+	t.Helper()
+	chainSelector, err := chainsel.SelectorFromChainId(cfg.ChainID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get chain selector from chain id: %w", err)
+	}
+	chain, ok := env.Chains[chainSelector]
+	if !ok {
+		return nil, fmt.Errorf("chain with id %d not found", cfg.ChainID)
+	}
+	_, deployedContract, err := changeset.DeployCapabilityRegistry(env, chainSelector)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deploy contract: %w", err)
+	}
+
+	nopsParams := v.NopsToNopsParams()
```

Comment:
@connorwstein it's not straightforward to reuse the existing changeset that deploys/configures our capability registry because it does too much stuff (and need to be refactored to support mcms as discussed in slack)

i'd like to move forward with this approach here so we can use the tool downstream

@vyzaldysanchez please file a ticket to refactor this logic into calls to bite-sizes changesets. and write in that ticket that we need to implement the changesets first.